tests/integration/pilot/common/routing.go

```diff
@@ -452,6 +452,46 @@ spec:
 	return []TrafficTestCase{tc}
 }
 
+// useClientProtocolCases contains tests use_client_protocol from DestinationRule
+func useClientProtocolCases(apps *EchoDeployments) []TrafficTestCase {
+	var cases []TrafficTestCase
+	client := apps.PodA
+	destination := apps.PodC[0]
+	cases = append(cases,
+		TrafficTestCase{
+			name:   "use client protocol with h2",
+			skip:   apps.IsMulticluster(), // Not needed for multi cluster.
+			config: useClientProtocolDestinationRule("use-client-protocol-h2", destination.Config().Service),
+			call:   client[0].CallWithRetryOrFail,
+			opts: echo.CallOptions{
+				Target:   destination,
+				PortName: "http",
```

Comment:
would be good to test calls to `http`, `http2` (not sure ifthis one exists yet), and `auto-http` as well to make sure it works in all cases.

and maybe the behavior with/without the DR

Reply:
Will do